storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -28,94 +32,270 @@
+                schedulingPrioritystrategy.prepare(this.topologies, this.cluster, this.userMap, this.nodes);
+                //Call scheduling priority strategy
+                td = schedulingPrioritystrategy.getNextTopologyToSchedule();
+            } catch (Exception e) {
+                LOG.error("Exception thrown when running priority strategy {}. No topologies will be scheduled! Error: {} StackTrace: {}"
+                        , schedulingPrioritystrategy.getClass().getName(), e.getMessage(), Arrays.toString(e.getStackTrace()));
+                break;
+            }
+            if (td == null) {
+                break;
+            }
+            scheduleTopology(td);
+        }
+    }
+
+    public void scheduleTopology(TopologyDetails td) {
+        User topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+        if (cluster.getUnassignedExecutors(td).size() > 0) {
+            LOG.debug("/********Scheduling topology {} from User {}************/", td.getName(), topologySubmitter);
+
+            SchedulingState schedulingState = this.checkpointSchedulingState();
+            IStrategy rasStrategy = null;
+            try {
+                rasStrategy = (IStrategy) Utils.newInstance((String) td.getConf().get(Config.TOPOLOGY_SCHEDULER_STRATEGY));
+            } catch (RuntimeException e) {
+                LOG.error("failed to create instance of IStrategy: {} with error: {}! Topology {} will not be scheduled.",
+                        td.getName(), td.getConf().get(Config.TOPOLOGY_SCHEDULER_STRATEGY), e.getMessage());
+                this.restoreCheckpointSchedulingState(schedulingState);
+                //since state is restored need the update User topologySubmitter to the new User object in userMap
+                topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                topologySubmitter.moveTopoFromPendingToInvalid(td);
+                this.cluster.setStatus(td.getId(), "Unsuccessful in scheduling - failed to create instance of topology strategy "
+                        + td.getConf().get(Config.TOPOLOGY_SCHEDULER_STRATEGY) + ". Please check logs for details");
+                return;
+            }
+            IEvictionStrategy evictionStrategy = null;
+            while (true) {
+                SchedulingResult result = null;
+                try {
+                    //Need to re prepare scheduling strategy with cluster and topologies in case scheduling state was restored
+                    rasStrategy.prepare(this.topologies, this.cluster, this.userMap, this.nodes);
+                    result = rasStrategy.schedule(td);
+                } catch (Exception e) {
+                    LOG.error("Exception thrown when running strategy {} to schedule topology {}. Topology will not be scheduled! Error: {} StackTrace: {}"
+                            , rasStrategy.getClass().getName(), td.getName(), e.getMessage(), Arrays.toString(e.getStackTrace()));
```

Comment:
Yeah, it should.  [link](http://www.slf4j.org/apidocs/org/slf4j/Logger.html#error%28java.lang.String,%20java.lang.Throwable%29)